Src/FluentAssertions/Collections/CollectionAssertions.cs

```diff
@@ -933,6 +933,84 @@ public AndConstraint<TAssertions> ContainInOrder(params object[] expected)
             return new AndConstraint<TAssertions>((TAssertions)this);
         }
 
+        /// <summary>
+        /// Asserts the current collection does not contain the specified elements in the exact same order, not necessarily consecutive.
+        /// </summary>
+        /// <remarks>
+        /// Elements are compared using their <see cref="object.Equals(object)" /> implementation.
+        /// </remarks>
+        /// <param name="unexpected">A <see cref="System.Array"/> with the unexpected elements.</param>
+        public AndConstraint<TAssertions> NotContainInOrder(params object[] unexpected)
+        {
+            return NotContainInOrder(unexpected, string.Empty);
+        }
+
+        /// <summary>
+        /// Asserts the current collection does not contain the specified elements in the exact same order, not necessarily consecutive.
+        /// </summary>
+        /// <remarks>
+        /// Elements are compared using their <see cref="object.Equals(object)" /> implementation.
+        /// </remarks>
+        /// <param name="unexpected">An <see cref="IEnumerable"/> with the unexpected elements.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> NotContainInOrder(IEnumerable unexpected, string because = "",
+            params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(unexpected, nameof(unexpected), "Cannot verify absence of ordered containment against a <null> collection.");
+
+            Execute.Assertion
```

Comment:
If somebody uses an `AssertionScope`, `FailWith` will not throw immediately and return a `bool`

Reply:
Does that mean we should also return from the method if subject is `null`?

Reply:
That has nothing to do with my comment. What I meant is that you need to avoid executing the remainder of that method if the first `FailWith` returns  `false`. 

If the subject itself is `null`, it should also fail the assertion.

Reply:
Pushed changes for this case, please check if that does what you expect it to.